infer/default_provider.go

```diff
@@ -0,0 +1,251 @@
+	dp.components = append(dp.components, components...)
+	return dp
+}
+
+// WithFunctions adds the given functions to the provider.
+func (dp *DefaultProvider) WithFunctions(functions ...InferredFunction) *DefaultProvider {
+	dp.functions = append(dp.functions, functions...)
+	return dp
+}
+
+// WithConfig adds the given config to the provider.
+func (dp *DefaultProvider) WithConfig(config InferredConfig) *DefaultProvider {
+	dp.config = config
+	return dp
+}
+
+// WithModuleMap adds the given module map to the provider.
+func (dp *DefaultProvider) WithModuleMap(moduleMap map[tokens.ModuleName]tokens.ModuleName) *DefaultProvider {
+	dp.moduleMap = moduleMap
+	return dp
+}
+
+func (dp *DefaultProvider) WithLanguageMap(languageMap map[string]any) *DefaultProvider {
+	dp.metadata.LanguageMap = languageMap
+	return dp
+}
+
+func (dp *DefaultProvider) WithDescription(description string) *DefaultProvider {
+	dp.metadata.Description = description
+	return dp
+}
+
+func (dp *DefaultProvider) WithDisplayName(displayName string) *DefaultProvider {
+	dp.metadata.DisplayName = displayName
+	return dp
+}
+
+func (dp *DefaultProvider) WithKeywords(keywords ...string) *DefaultProvider {
+	dp.metadata.Keywords = append(dp.metadata.Keywords, keywords...)
+	return dp
+}
+
+func (dp *DefaultProvider) WithHomepage(homepage string) *DefaultProvider {
+	dp.metadata.Homepage = homepage
+	return dp
+}
+
+func (dp *DefaultProvider) WithRepository(repository string) *DefaultProvider {
+	dp.metadata.Repository = repository
+	return dp
+}
+
+func (dp *DefaultProvider) WithPublisher(publisher string) *DefaultProvider {
+	dp.metadata.Publisher = publisher
+	return dp
+}
+
+func (dp *DefaultProvider) WithLogoURL(logoURL string) *DefaultProvider {
+	dp.metadata.LogoURL = logoURL
+	return dp
+}
+
+func (dp *DefaultProvider) WithLicense(license string) *DefaultProvider {
+	dp.metadata.License = license
+	return dp
+}
+
+func (dp *DefaultProvider) WithPluginDownloadURL(pluginDownloadURL string) *DefaultProvider {
+	dp.metadata.PluginDownloadURL = pluginDownloadURL
+	return dp
+}
+
+// WithName sets the provider name.
+func (dp *DefaultProvider) WithName(name string) *DefaultProvider {
+	dp.name = name
+	return dp
+}
+
+// WithVersion sets the provider version.
+func (dp *DefaultProvider) WithVersion(version string) *DefaultProvider {
+	dp.version = version
+	return dp
+}
+
+// Build constructs the provider options based on the current state of the default provider.
+func (dp *DefaultProvider) Build() *Options {
+	return &Options{
+		Metadata:   dp.metadata,
+		Resources:  dp.resources,
+		Components: dp.components,
+		Functions:  dp.functions,
+		Config:     dp.config,
+		ModuleMap:  dp.moduleMap,
+	}
+}
+
+// validate checks if the default provider configuration is valid.
+func (dp *DefaultProvider) validate() error {
+	switch {
+	case dp.name == "":
+		return fmt.Errorf("provider name is required")
+	case dp.version == "":
```

Comment:
I think providers work now with no version.

Reply:
Yes, `version` is now optional. We will inject a git tag version if this provider is git-hosted.